app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt

```diff
@@ -0,0 +1,146 @@
+package co.condorlabs.customcomponents.simplecamerax
+
+import android.app.Activity
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.Matrix
+import android.graphics.drawable.Animatable
+import android.os.Bundle
+import android.view.View
+import android.view.WindowManager
+import androidx.appcompat.app.AppCompatActivity
+import androidx.camera.core.ImageCapture
+import co.condorlabs.customcomponents.CAMERA_TAKE_PHOTO_PARAM
+import co.condorlabs.customcomponents.R
+import co.condorlabs.customcomponents.models.CameraConfig
+import co.condorlabs.customcomponents.simplecamerax.fragment.SimpleCameraXFragment
+import kotlinx.android.synthetic.main.activity_camera.*
+import java.io.ByteArrayOutputStream
+
+class CameraActivity : AppCompatActivity(), SimpleCameraXFragment.OnCameraXListener {
+
+    private lateinit var cameraFragment: SimpleCameraXFragment
+    private lateinit var cameraConfigObj: CameraConfig
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
+        setContentView(R.layout.activity_camera)
+        cameraFragment = simpleCameraXFragment as SimpleCameraXFragment
+        cameraConfigObj = intent.getParcelableExtra(CAMERA_CONFIG_OBJ_PARAM)
+            ?: throw NoSuchElementException("CameraConfig object was not provided")
+        setupScreenProperties()
+    }
+
+    private fun setupScreenProperties() {
+        with(cameraConfigObj) {
+            titleText?.let { cameraTitle?.text = it }
+            descriptionText?.let { capturePhotoDescription?.text = it }
+            cancelButtonText?.let { cancelPhoto?.text = it }
+            cropButtonText?.let { cropPhoto?.text = it }
+        }
+    }
+
+    override fun fragmentTextureViewLoaded() {
+        initCameraX()
+        cancelPhoto?.setOnClickListener {
+            resetLayout()
+        }
+    }
+
+    private fun initCameraX() {
+        cameraFragment.startCamera()
+        captureButton?.setOnClickListener {
+            (captureButton?.drawable as? Animatable)?.start()
+            cameraFragment.takePhoto(cameraConfigObj.urlToSavePhoto)
+        }
+    }
+
+    override fun onCaptureSuccess(bitmap: Bitmap) {
+        showImageBitmap(bitmap)
+    }
+
+    override fun onImageSaved(bitmap: Bitmap) {
+        showImageBitmap(bitmap)
+    }
+
+    private fun showImageBitmap(bitmap: Bitmap) {
+        val bitmapHeight = bitmap.height
+        val bitmapWidth = bitmap.width
+
+        if (bitmapWidth > bitmapHeight) {
```

Comment:
I think you should create a function that explains this conditional.

Reply:
Ok